requests/models.py

```diff
@@ -347,9 +347,9 @@ def prepare_url(self, url, params):
         url = url.lstrip()
 
         # Don't do any URL preparation for non-HTTP schemes like `mailto`,
-        # `data` etc to work around exceptions from `url_parse`, which
-        # handles RFC 3986 only.
-        if ':' in url and not url.lower().startswith('http'):
+        # `data`, `http+unix` etc to work around exceptions from `url_parse`,
+        # which handles RFC 3986 only.
+        if ':' in url and not url.lower().startswith(('http://', 'https://')):
```

Comment:
The original commit (b149be5) was to solve issues like @tiran's but may have been implemented a bit naively. It doesn't look like this would negatively affect the original intent of the conditional. I do think we can remove the `if ':' in url` portion of the conditional now though since that's covered in the new `startswith` tuple.